cypress/integration/core-opensearch-dashboards/opensearch-dashboards/workspace-plugin/mds_workspace_assets.spec.js

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { BASE_PATH } from '../../../../utils/base_constants';
+
+let sourceWorkspaceName = 'test_source_workspace';
+let targetWorkspaceName = 'test_target_workspace';
+
+let sourceWorkspaceId = '';
+let targetWorkspaceId = '';
+let datasourceId = '';
+
+const setupWorkspace = (workspaceName, datasourceId) => {
+  return cy
+    .createWorkspace({
+      name: workspaceName,
+      settings: {
+        ...(datasourceId ? { dataSources: [datasourceId] } : {}),
+      },
+    })
+    .then((value) => {
+      // load sample data
+      cy.loadSampleDataForWorkspace('ecommerce', value, datasourceId);
+      cy.wrap(value);
+    });
+};
+
+if (Cypress.env('WORKSPACE_ENABLED')) {
+  describe('Workspace assets', () => {
+    before(() => {
+      cy.deleteWorkspaceByName(sourceWorkspaceName);
+      cy.deleteWorkspaceByName(targetWorkspaceName);
+      if (Cypress.env('DATASOURCE_MANAGEMENT_ENABLED')) {
+        cy.createDataSourceNoAuth().then((result) => {
+          datasourceId = result[0];
+          expect(datasourceId).to.be.a('string').that.is.not.empty;
+          setupWorkspace(sourceWorkspaceName, datasourceId).then(
+            (value) => (sourceWorkspaceId = value)
+          );
+          setupWorkspace(targetWorkspaceName, datasourceId).then(
+            (value) => (targetWorkspaceId = value)
+          );
+        });
+      } else {
+        setupWorkspace(sourceWorkspaceName).then(
+          (value) => (sourceWorkspaceId = value)
+        );
+        setupWorkspace(targetWorkspaceName).then(
+          (value) => (targetWorkspaceId = value)
+        );
+      }
+    });
+
+    after(() => {
+      cy.deleteWorkspaceByName(sourceWorkspaceName);
+      cy.deleteWorkspaceByName(targetWorkspaceName);
+      sourceWorkspaceId = '';
+      targetWorkspaceId = '';
```

Comment:
shall we uninstall sample data when do test data clean up?

Reply:
Yeah, we should uninstall to remove the data index.